console/src/main/java/org/georchestra/console/ws/AreaController.java

```diff
@@ -0,0 +1,85 @@
+package org.georchestra.console.ws;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.util.StreamUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+
+/**
+ * A simple controller to serve an area.json file from your datadir.
+ */
+@Controller
+public class AreaController {
+
+    @Value("${AreasUrl:area.geojson}")
+    private String areasUrl;
+
+    @Value("${georchestra.datadir:/etc/georchestra}")
+    private String datadir;
+
+    /**
+     * serve a json file put in the datadir or redirect to a url if AreaUrl in the
+     * config is set to a url.
+     * 
+     * @return json or rediurect to the resource if area is an URL
+     * @throws IOException
+     */
+    @GetMapping(value = "/public/area.geojson", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseBody
+    public String serveArea(HttpServletResponse response) throws IOException {
+        // if arealUrl in config is an http endpoint, then send it directly.
+        if (isURL(areasUrl)) {
+            response.sendRedirect(areasUrl);
```

Comment:
the file is meant to be loaded by the client via XHR ? Will it work if it points to whatever other server ?

Reply:
If the url is located outside the georchestra domain, then a preflight request will be launch yes, but we cannot control that, we could download the geojson and send it, but we could miss an edge case, redirecting seems more clean to me. I think people have the choice here : host the file in the datadir or elsewhere.